gammapy/maps/tests/test_hpxmap.py

```diff
@@ -382,3 +382,26 @@ def test_plot_poly():
     m = HpxNDMap.create(binsz=10)
     with mpl_plot_check():
         m.plot(method="poly")
+
+def test_hpxmap_addition_subtraction():
+    map1 = HpxMap.create(nside=8, unit='cm2')
+    map2 = HpxMap.create(nside=8, unit='m2')
+
+    map1 += 1*u.cm**2
+    map2 += 1*u.cm**2
+    assert_allclose(map2.data, 1e-4)
+    map3 = map1 - map2
+    assert_quantity_allclose(map3.quantity, 0*u.cm**2)
```

Comment:
Please put the assert for unit separately:
https://docs.gammapy.org/0.8/development/howto.html#assert-convention

Especially here I'd like to see how that map unit changes, the test should establish this explicitly.